sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -133,8 +143,8 @@ public synchronized void close() throws IOException {
                 throw lastError;
             }
         } finally {
-            // if close() is called again, an exception will be thrown
             this.lastError = new IOException(Constants.STREAM_CLOSED);
+            isClosed = true;
```

Comment:
Based on the above discussion, we should set this to true only when `lastError` is `null` or falls under the category of "won't retry".